rllib/core/learner/learner.py

```diff
@@ -495,20 +613,97 @@ def register_metric(self, module_id: str, key: str, value: Any) -> None:
         self._metrics[module_id][key] = value
 
     def register_metrics(self, module_id: str, metrics_dict: Dict[str, Any]) -> None:
-        """Registers a several key/value metric pairs for loss and gradient stats.
+        """Registers several key/value metric pairs for loss- and gradient stats.
 
         Args:
             module_id: The module_id to register the metrics under. This may be
                 ALL_MODULES.
             metrics_dict: A dict mapping names of metrics to be registered (below the
                 given `module_id`) to the actual values of these metrics. Values might
                 also be tensor vars (e.g. from within a traced tf2 function).
+                These will be automatically converted to numpy values.
         """
         for key, value in metrics_dict.items():
             self.register_metric(module_id, key, value)
 
-    def get_weights(self, module_ids: Optional[Set[str]] = None) -> Mapping[str, Any]:
-        """Returns the weights of the underlying MultiAgentRLModule.
+    def get_optimizer(
```

Comment:
New APIs to more easily retrieve registered optimizer by their module_ids and names AND to compile parameter sub-dictionaries that only contain param_refs -> params key/value-paris of a particular optimizer.